nextjs-13/src/lib/handler/app.handler.ts

```diff
@@ -0,0 +1,30 @@
+import { User } from '@prisma/client';
+import { StatusCodes } from 'http-status-codes';
+import { NextResponse, NextRequest } from 'next/server';
+import { getToken } from 'next-auth/jwt';
+
+import { findUserById } from 'repositories/user.repository';
+
+export default async function appHandler(
+  req: NextRequest,
+  callback: (currentUser: User, body: unknown) => Promise<NextResponse>
+) {
+  try {
+    const token = await getToken({ req });
+    const currentUser = await findUserById(token?.userId as string);
+
+    if (!currentUser)
+      return NextResponse.json(
+        { message: 'Invalid token' },
+        { status: StatusCodes.UNAUTHORIZED }
+      );
+
+    return await callback(currentUser, req);
+  } catch (err) {
+    console.error('========>err : ', err);
```

Comment:
So you still want to keep this? 😅 Can't we at least only have it in the development environment?

Reply:
Ah no let me remove it, I just miss it 😄